server/src/main/java/server/formatting/FormattingProvider.java

```diff
@@ -0,0 +1,118 @@
+package server.formatting;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.lsp4j.DocumentFormattingParams;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import server.buildifier.Buildifier;
+import server.buildifier.BuildifierException;
+import server.buildifier.BuildifierFileType;
+import server.buildifier.FormatInput;
+import server.buildifier.FormatOutput;
+import server.utils.DocumentTracker;
+
+public class FormattingProvider {
+
+    private static final Logger logger = LogManager.getLogger(FormattingProvider.class);
+    private DocumentTracker documentTracker;
+    private Buildifier buildifier;
+
+    public FormattingProvider(DocumentTracker documentTracker, Buildifier buildifier) {
+        this.documentTracker = documentTracker;
+        this.buildifier = buildifier;
+    }
+
+
+    public CompletableFuture<List<? extends TextEdit>> getDocumentFormatting(DocumentFormattingParams params) {
+        logger.info("Invoked FormattingProvider");
+
+        FormatInput formatInput = new FormatInput();
+        String stringUri = params.getTextDocument().getUri();
+        logger.info("Formatting file from: " + stringUri);
+        File file = getFileFromUriString(stringUri);
+
+        if (file == null) {
+            // TODO: handle this problem
+            logger.info("Could not find file");
+            return CompletableFuture.completedFuture(new ArrayList<TextEdit>());
+        }
+
+        String name = file.getName();
+        logger.info("Formatting file: " + name);
+
+        if (name.equals("BUILD")) {
+            formatInput.setType(BuildifierFileType.BUILD);
+        } else if (name.equals("WORKSPACE")) {
+            formatInput.setType(BuildifierFileType.WORKSAPCE);
+        } else {
+            // This assumes that this code will not be called on any incompatible file types
+            formatInput.setType(BuildifierFileType.BZL);
```

Comment:
Probably shouldn't assume that a keyboard shortcut won't get called on an incompatible file type. 

So what happens here if i'm looking at a java source file and i accidentally trigger the Formatting?

Reply:
Ok yeah the develop branch builds for me, but i am unable to build your branch. 
It looks like your BUILD files are out dated?
Also it seems like some imports are missing. 

We should probably add a section in the github CD pipeline to run the build scripts now that we know that they can pass when the build scripts fail.

Reply:
Yeah good point. I'll change it so it just returns an empty response if the file is incompatible. Or we could send a popup explaining that the type of file cannot be formatted.